src/EditorFeatures/Core/Implementation/Peek/PeekableItemSourceProvider.cs

```diff
@@ -24,8 +24,8 @@ internal sealed class PeekableItemSourceProvider : IPeekableItemSourceProvider
 
         [ImportingConstructor]
         public PeekableItemSourceProvider(
-            IPeekableItemFactory peekableItemFactory,
-            IPeekResultFactory peekResultFactory,
+            [Import(AllowDefault = true)] IPeekableItemFactory peekableItemFactory,
+            [Import(AllowDefault = true)] IPeekResultFactory peekResultFactory,
```

Comment:
📝 This is intentionally not optional. By design, `PeekableItemSourceProvider` will be silently omitted from the catalog when `IPeekResultFactory` is not available. The type does not function correctly without it.

Reply:
I'm making the type work properly without it below, by just returning null if the part is null.